pkg/network/controllers/hotplug.go

```diff
@@ -46,19 +46,35 @@ func ApplyDynamicIfaceRequestOnVMI(
 		if shouldBeHotUnplug {
 			vmiIface := vmispec.LookupInterfaceByName(vmiSpecCopy.Domain.Devices.Interfaces, vmIface.Name)
 			vmiIface.State = v1.InterfaceStateAbsent
+		} else {
+			applyInterfaceStateInVMI(vmiSpecCopy, vmIface)
 		}
 	}
+
 	return vmiSpecCopy
 }
 
+func applyInterfaceStateInVMI(vmiSpec *v1.VirtualMachineInstanceSpec, vmIface v1.Interface) {
+	if vmIface.State == v1.InterfaceStateLinkDown ||
+		vmIface.State == v1.InterfaceStateLinkUp ||
+		vmIface.State == "" {
+		vmiIface := vmispec.LookupInterfaceByName(vmiSpec.Domain.Devices.Interfaces, vmIface.Name)
+		// I don't hotplug
+		if vmiIface != nil {
+			vmiIface.State = vmIface.State
+		}
+	}
+}
+
 func ClearDetachedInterfaces(
 	specIfaces []v1.Interface,
 	specNets []v1.Network,
 	statusIfaces map[string]v1.VirtualMachineInstanceNetworkInterface,
 ) ([]v1.Interface, []v1.Network) {
 	var ifaces []v1.Interface
 	for _, iface := range specIfaces {
-		if _, existInStatus := statusIfaces[iface.Name]; (existInStatus && iface.State == v1.InterfaceStateAbsent) ||
+		_, existInStatus := statusIfaces[iface.Name]
+		if (existInStatus && iface.State == v1.InterfaceStateAbsent) ||
```

Comment:
Please consider postponing this change to a follow-up.
This PR's scope is already quite big.